pydatastructs/strings/algorithms.py

```diff
@@ -158,3 +161,31 @@ def _rabin_karp(text, query):
             positions.append(i)
 
     return positions
+
+
+# Boyer-Moore Implementation for string pattern matching
+
+NO_OF_CHARS = 256
+
+def badCharHeuristic(string, size):
+	badChar = [-1]*NO_OF_CHARS
+	for i in range(size):
+		badChar[ord(string[i])] = i
+	return badChar
+
+def search(txt, pat):
```

Comment:
Search function is already there. You just have to make minor changes to add your algorithm.